### PR TITLE
allow numbers in reddit post_id

### PR DESCRIPTION
**Pull request by swoker**
# Description

Reddit Threads can have numbers in them, so the regex should allow that

# Issue Fixes

None

# Checklist:

- [x] I am pushing changes to the **develop** branch
- [x] I am using the recommended development environment
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have formatted and linted my code using python-black and pylint
- [x] I have cleaned up unnecessary files
- [x] My changes generate no new warnings
- [x] My changes follow the existing code-style
- [x] My changes are relevant to the project

# Any other information (e.g how to test the changes)

None

### CODE REVIEW

**Comment by JasonLovesDoggo** [comment]
The checker just checked against URIs...

File: utils/.config.template.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ random = { optional = true, options = [true,
     false,
 ], default = false, type = "bool", explanation = "If set to no, it will ask you a thread link to extract the thread, if yes it will randomize it. Default: 'False'", example = "True" }
 subreddit = { optional = false, regex = "[_0-9a-zA-Z]+$", nmin = 3, explanation = "what subreddit to pull posts from, the name of the sub, not the URL", example = "AskReddit", oob_error = "A subreddit name HAS to be between 3 and 20 characters" }
-post_id = { optional = true, default = "", regex = "^((?!://|://)[+a-zA-Z])*$", explanation = "Used if you want to use a specific post.", example = "urdtfx" }
+post_id = { optional = true, default = "", regex = "^((?!://|://)[+a-zA-Z0-9])*$", explanation = "Used if you want to use a specific post.", example = "urdtfx" }
```
Comment (Drugsosos) [inline utils/.config.template.toml:16]: why not just `\w`?

```suggestion
post_id = { optional = true, default = "", regex = "^((?!://|://)[+\w])*$", explanation = "Used if you want to use a specific post.", example = "urdtfx" }
```

Comment (swoker) [reply utils/.config.template.toml:16]: `/w` would allow `_` as well, which cannot be part of a post_id

Comment (Drugsosos) [reply utils/.config.template.toml:16]: good point, what about this?

```suggestion
post_id = { optional = true, default = "", regex = "^((?!://|://)[+\w][^_])*$", explanation = "Used if you want to use a specific post.", example = "urdtfx" }
```

**Comment by Drugsosos** [comment]
> The checker just checked against URIs...

Yeah, there is negative lookahead for `://` but it won't match if there are numbers in id
I noticed that problem , there can be numbers in id, for example, `w35ocq`

File: utils/.config.template.toml (hunk 1 above)
Comment (swoker) [reply utils/.config.template.toml:16]: could work, but imho `a-zA-Z0-9` is much easier to read

**Review by Drugsosos — APPROVED** [review]
👍🏻

**Pull request merged**